pkg/testers/node/node.go

```diff
@@ -0,0 +1,179 @@
+	TestArgs                    string `desc:"A space-separated list of arguments to pass to node e2e test."`
+	BoskosAcquireTimeoutSeconds int    `desc:"How long (in seconds) to hang on a request to Boskos to acquire a resource before erroring."`
+	BoskosLocation              string `desc:"If set, manually specifies the location of the boskos server. If unset and boskos is needed"`
+
+	// boskos struct field will be non-nil when the deployer is
+	// using boskos to acquire a GCP project
+	boskos *client.Client
+
+	// this channel serves as a signal channel for the hearbeat goroutine
+	// so that it can be explicitly closed
+	boskosHeartbeatClose chan struct{}
+}
+
+func NewDefaultTester() *Tester {
+	return &Tester{
+		SkipRegex:                   `\[Flaky\]|\[Slow\]|\[Serial\]`,
+		Runtime:                     "docker",
+		BoskosLocation:              "http://boskos.test-pods.svc.cluster.local.",
+		BoskosAcquireTimeoutSeconds: 5 * 60,
+	}
+}
+
+func (t *Tester) Execute() error {
+	fs, err := gpflag.Parse(t)
+	if err != nil {
+		return fmt.Errorf("failed to initialize tester: %v", err)
+	}
+
+	klog.InitFlags(nil)
+	fs.AddGoFlagSet(flag.CommandLine)
+
+	help := fs.BoolP("help", "h", false, "")
+	if err := fs.Parse(os.Args); err != nil {
+		return fmt.Errorf("failed to parse flags: %v", err)
+	}
+
+	if *help {
+		fs.SetOutput(os.Stdout)
+		fs.PrintDefaults()
+		return nil
+	}
+	if err := t.validateFlags(); err != nil {
+		return fmt.Errorf("failed to validate flags: %v", err)
+	}
+
+	// try to acquire project from boskos
+	if t.GCPProject == "" {
+		klog.V(1).Info("no GCP project provided, acquiring from Boskos ...")
+
+		boskosClient, err := boskos.NewClient(t.BoskosLocation)
+		if err != nil {
+			return fmt.Errorf("failed to make boskos client: %s", err)
+		}
+		t.boskos = boskosClient
+
+		resource, err := boskos.Acquire(
+			t.boskos,
+			gceProjectResourceType,
+			time.Duration(t.BoskosAcquireTimeoutSeconds)*time.Second,
+			t.boskosHeartbeatClose,
+		)
+
+		if err != nil {
+			return fmt.Errorf("init failed to get project from boskos: %s", err)
+		}
+		t.GCPProject = resource.Name
+		klog.V(1).Infof("got project %s from boskos", t.GCPProject)
+	}
+
+	defer func() {
+		if t.boskos != nil {
+			klog.V(1).Info("releasing boskos project")
+			err := boskos.Release(
+				t.boskos,
+				t.GCPProject,
+				t.boskosHeartbeatClose,
+			)
+			if err != nil {
+				klog.Errorf("failed to release boskos project: %v", err)
+			}
+		}
+	}()
+
+	return t.Test()
+}
+
+func (t *Tester) validateFlags() error {
+	if t.RepoRoot == "" {
+		return fmt.Errorf("required --repo-root")
+	}
+	if t.GCPZone == "" {
+		return fmt.Errorf("required --gcp-zone")
+	}
+	return nil
+}
+
+func (t *Tester) constructArgs() []string {
+	defaultArgs := []string{
+		"REMOTE=true",
+		"DELETE_INSTANCES=true",
+	}
+
+	argsFromFlags := []string{
+		"SKIP=" + t.SkipRegex,
+		"FOCUS=" + t.FocusRegex,
+		"RUNTIME=" + t.Runtime,
```

Comment:
I'm not sure if just exposing `RUNTIME` is enough. For example, take this job: https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes/sig-node/crio.yaml#L52

```
--node-test-args=--container-runtime=remote --container-runtime-endpoint=unix:///var/run/crio/crio.sock ...
```

I'm not clear how that translates into equivalent args here...

Reply:
Note that power users can still set the corrsponding ENVs before invoking kubetest2 (until kubetest2 supports them as flags (if at all))

Reply:
> Note that power users can still set the corrsponding ENVs before invoking kubetest2

How does that work? Doesn't the make cmd execution need to inherit the bash env variables, i.e. something like `cmd.Env = os.Environ()`.

Additionally, I think the challenge is it's not clear what exact envs you need set because the current jobs using kubetest(1) node e2e runner took in more or less same args as [run_remote.go](https://github.com/kubernetes/kubernetes/blob/master/test/e2e_node/runner/remote/run_remote.go). However since this is based on `test-e2e-node.sh`, for all those jobs, you'll need to translate those args to the equivalent envs from `test-e2e-node` which isn't super straightforward.

Reply:
https://golang.org/pkg/os/exec/#Cmd

cmd will inherit all exported envs from parent process by default.

> I think the challenge is it's not clear what exact envs you need set

Ack, but I think that is orthogonal to the kubetest2 tester invocation. i.e. they come more from https://github.com/kubernetes/kubernetes/blob/96be00df69390ed41b8ec22facc43bcbb9c88aae/build/root/Makefile#L206-L271 

Ideally, we will port over all relevant functionality as flags over time.

I think the main question at hand is whether or not `sig-node` thinks `make test-e2e-node` is a better common entrypoint for node testing as opposed what we have currently https://github.com/kubernetes/test-infra/blob/4be647a393b6b491a2c0f00145d8bb9f8416b082/kubetest/e2e.go#L556-L611

Reply:
ack got it.

> I think the main question at hand is whether or not sig-node thinks make test-e2e-node is a better common entrypoint for node testing as opposed what we have currently

Yup, this is the main question. 

As I mentioned the earlier, the problem I see is if the goal is for us is to migrate all e2e node jobs to kubetest2 eventually this will basically require rewriting all of the job configs since the current kubetest1 interface doesn't rely on `make test-e2e-node`. On other side standardizing on `make test-e2e-node` maybe a good move eventually :) but worth to discuss.   

Folks in the [sig-node-ci group](https://docs.google.com/document/d/1fb-ugvgdSVIkkuJ388_nhp2pBTy_4HEVg5848Xy7n5U) would be candidates to ask about this. 

cc a couple folks to get their thoughts on usage of `make test-e2e-node` vs existing kubetest node e2e interface.

/cc @SergeyKanzhelev @ehashman @fromanirh @knabben

Reply:
+1 for standardizing on `make test-e2e-node`